### PR TITLE
GTM: Better sync run decisions 

### PR DESCRIPTION
**Pull request by gfiorav**
Please CR @javitonino

### CODE REVIEW

File: lib/carto/ghost_tables_manager.rb [hunk 1]
```diff
@@ -41,11 +42,15 @@ def user_tables_synced_with_db?
...
+    # It's nice to run sync if any unsafe stale (dropped or renamed) tables will be shown to the user but we can't block
+    # the workers for more that 180 seconds
+    def safe_sync?
       cartodbfied_tables = fetch_cartodbfied_tables
 
-      find_dropped_tables(cartodbfied_tables).empty? && find_stale_tables(cartodbfied_tables).empty?
+      dropped_and_stale_table_count = find_dropped_tables(cartodbfied_tables).count +
+                                      find_stale_tables(cartodbfied_tables).count
+
+      dropped_and_stale_table_count < MAX_TABLES_FOR_SYNC_RUN
```
Comment (javitonino) [inline lib/carto/ghost_tables_manager.rb]: This always run synchronously if there are 0 stale/dropped tables, doesn't it? Even if there are lots of new tables.

I think the check we want is to only run sync if "there are any stale tables" and "there are less than X tables to operate on (including linking new tables)"

Comment (gfiorav) [inline lib/carto/ghost_tables_manager.rb]: yeah, mindless overlook, give me a sec

**Comment by javitonino** [comment]
LGTM
:package: :it: after tests

**Comment by gfiorav** [comment]
Thanks for the review and deploying this!

Hope it's smooth !

**Pull request merged**